app/controllers/oaisys/pmh_controller.rb

```diff
@@ -73,6 +73,7 @@ def list_metadata_formats
       raise Oaisys::NoMetadataFormatsError.new(parameters: parameters) if formats.empty?
     end
 
+    parameters['identifier'].prepend('oai:era.library.ualberta.ca:') if parameters['identifier'].present?
```

Comment:
You might as well wrap this in a little method or something you can call, since you have to do it in a few places.

```ruby
render :list_metadata_formats, formats: :xml, locals: { formats: formats, parameters: prep_identifiers(parameters) }
```

basically (but use a better name than that, that's awful)